01-js/easy/expenditure-analysis.js

```diff
@@ -9,7 +9,21 @@
 */
 
 function calculateTotalSpentByCategory(transactions) {
-  return [];
+  let categoryObject = {};
+  for ( let i=0 ; i<transactions.length; i++) {
+    let category = transactions[i].category;
+    let price  = transactions[i].price;
+    if (categoryObject[category]) {
+      categoryObject[category] += price;
+    } else {
+      categoryObject[category] = price;
+    }
+  }
+  let ans = Object.keys(categoryObject).map((key) => {
+    return { category: key, totalSpent: categoryObject[key]}
+  })
```

Comment:
```js
let ans = Object.keys(categoryObject).map((category) => ({
  category,
  totalSpent: categoryObject[category],
}));
```

Reply:
> ```js
> let ans = Object.keys(categoryObject).map((category) => ({
>   category,
>   totalSpent: categoryObject[category],
> }));
> ```

Thank you so much for your to help me improve

Reply:
> ```js
> let ans = Object.keys(categoryObject).map((category) => ({
> ```

Thank you so much for your time to help me improve